src/ReplaceReferencesCapableTrait.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Dhii\Config;
+
+use InvalidArgumentException;
+use Psr\Container\ContainerInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+
+/**
+ * Trait ReplaceReferencesCapableTrait.
+ *
+ * Adds ability to replace all tokens in the form `${key}` in a string with corresponding values retrieved from a container.
+ *
+ * @since [*next-version*]
+ */
+trait ReplaceReferencesCapableTrait
+{
+    /**
+     * Replaces all tokens in the form `${key}` in a string with corresponding values retrieved from a container.
+     *
+     * @since [*next-version*]
+     *
+     * @param string             $input
+     * @param ContainerInterface $container
+     * @param string             $startDelimiter
+     * @param string             $endDelimiter
+     *
+     * @throws \Psr\Container\ContainerExceptionInterface
+     * @throws \Psr\Container\NotFoundExceptionInterface
+     *
+     * @return string The resulting string.
+     */
+    protected function _replaceReferences($input, ContainerInterface $container, $startDelimiter = '${', $endDelimiter = '}')
+    {
+        $subject = $this->_normalizeString($input);
+
+        preg_match_all('/\\' . $startDelimiter . '[^\\' . $endDelimiter . ']+' . $endDelimiter . '/', $subject, $matches);
```

Comment:
`[^\\]` means "anything excluding backslash. Why?

Reply:
Because without `\\` it will escape `\'`

Reply:
`'[^\'` - syntax error

Reply:
The question was not "Why did you escape the backslash?", but "Why anything excluding backslash?".

Reply:
I rewrote regex, so it is more clean now.

Reply:
@romalytvynenko, I still don't know why you allow "anything excluding backslash" in that char class.